frontend/public/components/monitoring/promql-expression-input.tsx

```diff
@@ -0,0 +1,395 @@
+import {
+  autocompletion,
+  closeCompletion,
+  completionKeymap,
+  currentCompletions,
+  setSelectedCompletion,
+} from '@codemirror/autocomplete';
+import { closeBrackets, closeBracketsKeymap } from '@codemirror/closebrackets';
+import { defaultKeymap, insertNewlineAndIndent } from '@codemirror/commands';
+import { HighlightStyle, tags } from '@codemirror/highlight';
+import { history, historyKeymap } from '@codemirror/history';
+import { indentOnInput } from '@codemirror/language';
+import { lintKeymap } from '@codemirror/lint';
+import { bracketMatching } from '@codemirror/matchbrackets';
+import { highlightSelectionMatches } from '@codemirror/search';
+import { EditorState, Prec } from '@codemirror/state';
+import {
+  EditorView,
+  highlightSpecialChars,
+  keymap,
+  placeholder as codeMirrorPlaceholder,
+  ViewPlugin,
+  ViewUpdate,
+} from '@codemirror/view';
+import { YellowExclamationTriangleIcon } from '@console/shared';
+import CloseButton from '@console/shared/src/components/close-button';
+import { PromQLExtension } from 'codemirror-promql';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { PROMETHEUS_BASE_PATH } from '../graphs';
+import { PrometheusEndpoint } from '../graphs/helpers';
+import { useSafeFetch } from '../utils';
+import './_promql-expression-input.scss';
+
+interface PromQLExpressionInputProps {
+  value: string;
+  onValueChange: (value: string) => void;
+  onExecuteQuery?: () => void;
+}
+
+const promqlExtension = new PromQLExtension();
+
+export const theme = EditorView.theme({
+  '.cm-scroller': {
+    fontFamily: 'inherit',
+  },
+  '&': {
+    '&.cm-editor.cm-focused': {
+      outline: 'none',
+    },
+  },
+  '.cm-tooltip.cm-completionInfo': {
+    backgroundColor: 'var(--pf-global--palette--blue-50)',
+    border: 'none',
+    marginTop: '-11px',
+    padding: '10px',
+  },
+  '.cm-completionInfo.cm-completionInfo-right': {
+    '&:before': {
+      content: "' '",
+      height: '0',
+      position: 'absolute',
+      width: '0',
+      left: '-20px',
+      borderWidth: '10px',
+      borderStyle: 'solid',
+      borderColor: 'transparent',
+      borderRightColor: 'var(--pf-global--palette--blue-50)',
+    },
+    marginLeft: '12px',
+  },
+  '.cm-completionInfo.cm-completionInfo-left': {
+    '&:before': {
+      content: "' '",
+      height: '0',
+      position: 'absolute',
+      width: '0',
+      right: '-20px',
+      borderWidth: '10px',
+      borderStyle: 'solid',
+      borderColor: 'transparent',
+      borderLeftColor: 'var(--pf-global--palette--blue-50)',
+    },
+    marginRight: '12px',
+  },
+  '.cm-completionIcon': {
+    fontFamily: 'codicon',
+    width: '1.5em',
+    verticalAlign: 'middle',
+  },
+  '.cm-selectionMatch': {
+    backgroundColor: '#E7F1F4',
```

Comment:
Nit: We're using lower case for color codes elsewhere

Reply:
@jgbernalp Have you tested this with dark mode? See https://issues.redhat.com/browse/CONSOLE-3048

cc @mattnolting @sg00dwin

Reply:
I rebased to get the dark mode changes, will adjust the colors

Reply:
@kyoto I changed to use Patternfly css vars so the color os not hardcoded

Reply:
https://user-images.githubusercontent.com/5461414/160798978-dd4a3102-5000-40cc-b011-5d5b37152833.mov